tests/test_local_env.py

```diff
@@ -58,23 +57,62 @@ def test_determine_package_dir__various_paths_in_poetry2nix_env(
         }
     )
     path = tmp_path / subdir
-    expect = tmp_path / f"lib/python{major}.{minor}/site-packages"
-    assert LocalPackageResolver.determine_package_dir(path) == expect
+    expect = {tmp_path / f"lib/python{major}.{minor}/site-packages"}
+    assert set(LocalPackageResolver.find_package_dirs(path)) == expect
 
 
 @pytest.mark.parametrize(
     "subdir", [pytest.param(d, id=f"pep582:{d}") for d in pep582_subdirs]
 )
-def test_determine_package_dir__various_paths_in_pypackages(write_tmp_files, subdir):
+def test_find_package_dirs__various_paths_in_pypackages(write_tmp_files, subdir):
     # A directory structure that resembles a minimal PEP582 __pypackages__ dir:
     tmp_path = write_tmp_files(
         {
             f"__pypackages__/{major}.{minor}/lib/some_package.py": "",
         }
     )
     path = tmp_path / subdir
-    expect = tmp_path / f"__pypackages__/{major}.{minor}/lib"
-    assert LocalPackageResolver.determine_package_dir(path) == expect
+    expect = {tmp_path / f"__pypackages__/{major}.{minor}/lib"}
+    assert set(LocalPackageResolver.find_package_dirs(path)) == expect
+
+
+@pytest.mark.parametrize(
+    "subdir",
+    [pytest.param(d, id=f"pep582:{d}") for d in pep582_subdirs]
+    + [pytest.param(".venv/" + d, id=f"venv:.venv/{d}") for d in env_subdirs],
+)
+def test_find_package_dirs__envs_with_multiple_package_dirs(write_tmp_files, subdir):
+    # A directory structure that contains multiple Python environments, and
+    # multiple package dirs inside each Python environments:
+    tmp_path = write_tmp_files(
+        {
+            f"__pypackages__/{major}.{minor}/lib/first_package.py": "",
+            f"__pypackages__/{major}.{minor + 1}/lib/second_package.py": "",
+            ".venv/bin/python": "",
+            f".venv/lib/python{major}.{minor}/site-packages/third_package.py": "",
+            f".venv/lib/python{major}.{minor + 1}/site-packages/fourth_package.py": "",
+        }
+    )
+    path = tmp_path / subdir
+    actual = set(LocalPackageResolver.find_package_dirs(path))
+
+    def expected_package_dirs(base: Path, subdir: str) -> Iterator[Path]:
+        # When given:
+        # __pypackages__/  -->  return all .../X.Y/lib dirs
+        # __pypackages__/X.Y/*  -->  return the only matching .../lib dir
+        # .venv/  -->  return all .../lib/pythonX.Y/site-packages dirs
```

Comment:
There is a mismatch between the description and the code. There is an assumption that `all` `.../X.Y/lib` dirs mean `{major}.{minor}` and `{major}.{minor+1}`. It may be enough to rewrite the code below as suggested in the comment below and remove the description if you fid the suggestion useful.

Reply:
I wrote both the comment and the test with the assumption that `X.Y` refers to _any_ Python version, no matter what the _current_ Python version happens to be. To that end I use `{major}.{minor}` and `{major}.{minor+1}` to mean "two different Python version, one that happens to match the current version and one that doesn't".

Then, in the parametrized inputs (ultimately from `*_subdirs` above) there are both paths that _do_ include `{major}.{minor}` and paths that _don't_. For those that _do_, we expect only the _one_ matching package dir to be returned, and for those that _don't_ we expect _both_ package dirs to be returned.

I'm not sure I see the mismatch between the comment and the code, but it probably makes sense to drop the comment and make the code more readable instead.

Reply:
So maybe just drop this comment?

My point was that realisation of a more general `{major}.{minor1}` and `{major}.{minor2}` is done here with `major2 = major1 + 1`. Which is OK, but then the comment does not match.

Reply:
Dropped in e0d7c83.